src/main/java/com/okta/tools/OktaAwsCliAssumeRole.java

```diff
@@ -36,46 +36,71 @@
 
     private static final Logger logger = LogManager.getLogger(OktaAwsCliAssumeRole.class);
 
+    private OktaAwsCliEnvironment environment;
+
+    private SessionHelper sessionHelper;
+    private ConfigHelper configHelper;
+    private RoleHelper roleHelper;
+    private ProfileHelper profileHelper;
+
+    private OktaSaml oktaSaml;
+
+    private Optional<Session> currentSession;
+    private Optional<Profile> currentProfile;
+
     static OktaAwsCliAssumeRole createOktaAwsCliAssumeRole(String oktaOrg, String oktaAWSAppURL, String oktaAWSUsername, String oktaAWSPassword, String oktaProfile, String oktaAWSRoleToAssume) {
         return new OktaAwsCliAssumeRole(oktaOrg, oktaAWSAppURL, oktaAWSUsername, oktaAWSPassword, oktaProfile, oktaAWSRoleToAssume);
     }
 
     private OktaAwsCliAssumeRole(String oktaOrg, String oktaAWSAppURL, String oktaUsername, String oktaAWSPassword, String oktaProfile, String awsRoleToAssume) {
-        OktaAwsCliEnvironment.oktaOrg = oktaOrg;
-        OktaAwsCliEnvironment.oktaAwsAppUrl = oktaAWSAppURL;
-        OktaAwsCliEnvironment.oktaUsername = oktaUsername;
-        OktaAwsCliEnvironment.oktaPassword = oktaAWSPassword;
-        OktaAwsCliEnvironment.oktaProfile = oktaProfile;
-        OktaAwsCliEnvironment.awsRoleToAssume = awsRoleToAssume;
+        environment = new OktaAwsCliEnvironment(oktaOrg, oktaUsername, oktaAWSPassword, oktaProfile, oktaAWSAppURL, awsRoleToAssume);
+    }
+
+    private void init() throws Exception {
+        sessionHelper = new SessionHelper(environment);
+        configHelper = new ConfigHelper(environment);
+        roleHelper = new RoleHelper(environment);
+        profileHelper = new ProfileHelper(environment);
+
+        oktaSaml = new OktaSaml(environment);
+
+        currentSession = sessionHelper.getCurrentSession();
+
+        if (currentSession.isPresent()) {
+            environment.oktaProfile = currentSession.get().profileName;
```

Comment:
After this point, **environment.oktaProfile** is not set to the correct value for the profile specified by **OKTA_PROFILE**. As a result, multiple profile selection doesn't work.

Right now, the named profile support doesn't work if there is also a session file. The expected behavior is if **OKTA_PROFILE** is not blank, then multiple-profile support should be used to pick the session. Only if it is blank should the legacy session file be used. The way the old system achieved this was to overwrite the legacy session file after determining the profile name from named multi-profile support.

There really should be more unit tests to cover these cases. Sorry about that.

Please try some tests with OKTA_PROFILE set to different values. You should be able to switch between the profiles without re-authenticating, and create new profiles only by seeing the role selection menu.

```bash
env OKTA_PROFILE=dev withokta sts get-caller-identity
env OKTA_PROFILE=prod awscli sts get-caller-identity
env OKTA_PROFILE=dev awscli sts get-caller-identity
env OKTA_PROFILE=prod awscli sts get-caller-identity
```

Make sure you have two different roles to assume. Expect the Okta authentication only once, and the role selection to appear twice times. Check that the get-caller-identity returns the expected assumed-role ARN.

Example:
```plain
arn:aws:sts::123456789012:assumed-role/Admin/botocore-session-1234567890
```